src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -783,77 +799,71 @@ private static BinderCheckTypeResult CheckComparisonArgTypesCore(IErrorContainer
             var coercions = new List<BinderCoercionResult>();
             var usePowerFxV1CompatibilityRules = features.PowerFxV1CompatibilityRules;
 
-            // Special case for number-backed BuiltIn Option Sets:
+            // Special cases for comparing option set values
             if (typeLeft.Kind == DKind.OptionSetValue || typeRight.Kind == DKind.OptionSetValue)
             {
-                // Option set must be numeric to compare. 
-                if ((typeLeft.OptionSetInfo?.BackingKind ?? DKind.Number) != DKind.Number)
-                {
-                    errorContainer.EnsureError(
-                        DocumentErrorSeverity.Severe,
-                        left.Parent,
-                        TexlStrings.ErrUnOrderedTypeForComparison_Type,
-                        typeLeft.GetKindString());
-
-                    return new BinderCheckTypeResult();
-                }
-                else if ((typeRight.OptionSetInfo?.BackingKind ?? DKind.Number) != DKind.Number)
+                // Comparing values from two different option set values is not supported, with or without V1 compat rules
```

Comment:
>// Comparing values from two different option set values is not supported, with or without V1 compat rules [](http://example.com/codeflow?start=16&length=106)

Today you can compare two different enums (e.g., `ImageRotation.None = GridStyle.None`, `JSONFormat.IgnoreUnsupportedTypes = TraceOptions.IgnoreUnsupportedTypes`). Will this break?